### PR TITLE
add Makefile and checkprerequisites

### PR DESCRIPTION
**Pull request by Dreat**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+test: node_modules lint unit
+
+report: node_modules unit-coverage open-coverage-report
+
+docker:
+	docker build -t transaction_tagger -f ./Dockerfile . 
+
+
+.PHONY: lint unit e2e spin-up-development-server open-coverage-report unit-coverage
```
Comment (allgreed) [inline Makefile]: no e2e in targets :c

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+.DEFAULT_GOAL := serve
+
+CHECK_PREREQUISITES := $(shell ./check-prerequisites)
+ifneq ($(CHECK_PREREQUISITES),)
+$(error $(CHECK_PREREQUISITES))
+endif
+
+
+.PHONY: serve build test report docker
```
Comment (allgreed) [inline Makefile]: no build :c

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+docker:
+	docker build -t transaction_tagger -f ./Dockerfile . 
+
+
+.PHONY: lint unit spin-up-development-server open-coverage-report unit-coverage
+lint:
+	yarn run lint
+unit:
+	yarn run test:unit
```
Comment (allgreed) [inline Makefile]: The command is `test` not `test:unit`

**Comment by LiquidLemon** [comment]
I'm not really sure this is necessary and it makes `make` a build dependency (which is not common for JavaScript projects).

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+.PHONY: serve build test report docker
+serve: node_modules spin-up-development-server
+
+test: node_modules lint unit
+
+report: node_modules unit-coverage open-coverage-report
+
+docker:
+	docker build -t transaction_tagger -f ./Dockerfile .
```
Comment (allgreed) [inline Makefile]: The tag should be `spotty-kiwi`

**Pull request merged**